lms/djangoapps/program_enrollments/tasks.py

```diff
@@ -0,0 +1,60 @@
+""" Tasks for program enrollments """
+from datetime import timedelta
+import logging
+from celery import task
+from celery_utils.logged_task import LoggedTask
+from django.utils import timezone
+
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment, ProgramCourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+@task(base=LoggedTask)
+def expire_waiting_enrollments(expiration_days):
+    """
+    Remove all program_enrollments and related program_course_enrollments for
+    program_enrollments not modified for <expiration_days>
+    """
+    expiry_date = timezone.now() - timedelta(days=expiration_days)
+
+    program_enrollments = ProgramEnrollment.objects.filter(
+        user=None,
+        modified__lte=expiry_date
+    ).prefetch_related('program_course_enrollments')
```

Comment:
Nit: maybe `select_related` instead.

Reply:
ah yup, that's what I actually wanted

Reply:
err actually scratch that.  It's a reverse FK so select_related won't work

Reply:
ah that's right - ignore my similar comment in your other PR, then :)